Sources/ComposedUI/CollectionView/CollectionCoordinator.swift

```diff
@@ -217,6 +217,14 @@ extension CollectionCoordinator: SectionProviderMappingDelegate {
     public func mappingWillBeginUpdating(_ mapping: SectionProviderMapping) {
         reset()
         defersUpdate = true
+
+        // This is called here to ensure that the collection view's internal state is in-sync with the state of the
+        // data in hierarchy of sections. If this is not done it can cause various crashes when `performBatchUpdates` is called
+        // due to the collection view requesting data for sections that no longer exist, or crashes because the collection view is
+        // told to delete/insert from/into sections that it does not yet think exist.
+        //
+        // For more information on this see https://github.com/composed-swift/ComposedUI/pull/14
+        collectionView.layoutIfNeeded()
```

Comment:
Doesn't this break animations?

Reply:
Maybe if there are pending changes but from what I can tell this only happens when changes occur when the view is not visible or `reloadData` is used. In both cases I don't think animation would normally occur?

It doesn't break general animations for inserts/updates/deletes.

For what it's worth it looks like IGListKit does this too: https://github.com/Instagram/IGListKit/blob/02ffb1163bbd6dcbae4b33c127b47012c5c06741/Source/IGListKit/Internal/IGListBatchUpdateTransaction.m#L178

Reply:
Ok cool!